### PR TITLE
Frontend requests should not set cookies. Remove them, to allow Varnish to do a better job of caching the request.

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

File: src/Controllers/Frontend.php [hunk 1]
```diff
@@ -70,6 +70,32 @@ public function before(Request $request, Application $app)
...
+    public function after(Request $request, Response $response, Application $app)
+    {
+        // Frontend requests should not set cookies. Remove them, to allow Varnish
+        // (or reverse-proxies) to do a better job of caching the request.
+        if (!headers_sent()) {
+            $headersList = headers_list();
+            foreach($headersList as $header) {
+                if (strpos($header, "Set-Cookie: bolt_session=") === 0) {
+                    header_remove("Set-Cookie");
+                }
```
Comment (GawainLynch) [inline src/Controllers/Frontend.php]: Would this be simpler:
```php
$response->headers->clearCookie('bolt_session');
```

Comment (bobdenotter) [inline src/Controllers/Frontend.php]: It would, if i had known about it. :-)

Comment (GawainLynch) [inline src/Controllers/Frontend.php]: It aslo does the [headers_sent()](https://github.com/symfony/HttpFoundation/blob/2.6/Response.php#L332)

**Comment by bobdenotter** [comment]
PR updated.

**Comment by bobdenotter** [comment]
Fixed for 2.3 by #3564

**Pull request closed without merging**